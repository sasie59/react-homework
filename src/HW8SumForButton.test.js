import React from 'react';
import { getByTestId, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW8SumForButton';

test('should HW6RemovableList', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should sum = sum + number', () => {
  const { container, getByTestId } = render(<HW />);
  const div = container.querySelector('div[name="sum"]');
  userEvent.click(getByTestId(0));
  userEvent.click(getByTestId(2));

  expect(div.textContent).toBe('加總值 : 4');
  expect(container).toMatchSnapshot();
});
test('should sum = sum + number', () => {
  const { container, getByTestId } = render(<HW />);
  const div = container.querySelector('div[name="sum"]');
  userEvent.click(getByTestId(8));
  userEvent.click(getByTestId(6));

  expect(div.textContent).toBe('加總值 : 16');
  expect(container).toMatchSnapshot();
});
