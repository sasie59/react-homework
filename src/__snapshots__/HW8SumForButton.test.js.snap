// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW6RemovableList 1`] = `
<div>
  <div>
    <h1>
      HW8SumForButton
    </h1>
    <button
      data-testid="0"
    >
      1
    </button>
    <button
      data-testid="1"
    >
      2
    </button>
    <button
      data-testid="2"
    >
      3
    </button>
    <button
      data-testid="3"
    >
      4
    </button>
    <button
      data-testid="4"
    >
      5
    </button>
    <button
      data-testid="5"
    >
      6
    </button>
    <button
      data-testid="6"
    >
      7
    </button>
    <button
      data-testid="7"
    >
      8
    </button>
    <button
      data-testid="8"
    >
      9
    </button>
    <div
      name="sum"
    >
      加總值 : 
      0
    </div>
  </div>
</div>
`;

exports[`should sum = sum + number 1`] = `
<div>
  <div>
    <h1>
      HW8SumForButton
    </h1>
    <button
      data-testid="0"
    >
      1
    </button>
    <button
      data-testid="1"
    >
      2
    </button>
    <button
      data-testid="2"
    >
      3
    </button>
    <button
      data-testid="3"
    >
      4
    </button>
    <button
      data-testid="4"
    >
      5
    </button>
    <button
      data-testid="5"
    >
      6
    </button>
    <button
      data-testid="6"
    >
      7
    </button>
    <button
      data-testid="7"
    >
      8
    </button>
    <button
      data-testid="8"
    >
      9
    </button>
    <div
      name="sum"
    >
      加總值 : 
      4
    </div>
  </div>
</div>
`;

exports[`should sum = sum + number 2`] = `
<div>
  <div>
    <h1>
      HW8SumForButton
    </h1>
    <button
      data-testid="0"
    >
      1
    </button>
    <button
      data-testid="1"
    >
      2
    </button>
    <button
      data-testid="2"
    >
      3
    </button>
    <button
      data-testid="3"
    >
      4
    </button>
    <button
      data-testid="4"
    >
      5
    </button>
    <button
      data-testid="5"
    >
      6
    </button>
    <button
      data-testid="6"
    >
      7
    </button>
    <button
      data-testid="7"
    >
      8
    </button>
    <button
      data-testid="8"
    >
      9
    </button>
    <div
      name="sum"
    >
      加總值 : 
      16
    </div>
  </div>
</div>
`;
